src/go/k8s/.golangci.yml

```diff
@@ -0,0 +1,132 @@
+linters-settings:
```

Comment:
@ivotron - we should move this to  `src/go/` after this gets merged and enable the same linting throughout.